### ISSUE TITLE
[accordion] h3 as an Accordion Header issue

### ISSUE THREAD

**Issue by kotAPI** [mui/base-ui#1678]
Just wanted to understand how Base UI is solving the semantic issue related to Heading hierarchy, I've noticed a similar pattern that[ Radix uses too](https://github.com/radix-ui/primitives/issues/2738#issuecomment-1969364180) following the WCAG's example

In the WCAG pattern documentation, it is stated that they had used h3 in the example to match their heading hierarchy based on their structure to match the docs and get their heading hierarchy right so it makes sense for them to use the `<h3>` tag
https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/
<img width="979" alt="Image" src="https://github.com/user-attachments/assets/9966a260-0a61-40ed-934d-f0c7b98b5a91" />

When you inspect the Base UI's markup for the [Accordion example](https://base-ui.com/react/components/accordion), you see that this is off (and not accessible) as there is no h2 present as the markup directly jumps from h1 to h3

<img width="779" alt="Image" src="https://github.com/user-attachments/assets/9944df3c-e574-40c3-b82d-2982354d7ae9" />

### Proposal

I guess it would make sense to use a div as a safe default and let people override this based on their markup as and when required via `render`?

**Comment by colmtuite** [comment]
Closing this for now since I think the default is good, and changing it for the hero demo would require using the `render` prop, which is kinda outside the scope of the hero demo. Imo, demos don't necessarily count as part of the document in the usual sense, there's even an argument they should render inside an `<iframe>`.

**Comment by kotAPI** [comment]
> Closing this for now since I think the default is good, and changing it for the hero demo would require using the `render` prop, which is kinda outside the scope of the hero demo. Imo, demos don't necessarily count as part of the document in the usual sense, there's even an argument they should render inside an `<iframe>`.

Sure, the original problem still stands - like you mentioned - using render prop to override the default h3 is out of scope of this example, your users would have the same pain points in correcting their markup due to the defaults chosen here as they are forced to use `render` prop where Accordion is used.

**Comment by mj12albert** [comment]
> I guess it would make sense to use a div as a safe default and let people override this based on their markup as and when required via `render`?

I think it's impossible to pick a default heading level that will satisfy everybody, but a heading element with the incorrect level in a particular use-case is easy to spot, whereas defaulting to a `div` would require everybody to render their own heading element in all cases
